app/components/BuyModal/index.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { FC, useCallback, useEffect, useState } from "react";
+import { useAppDispatch } from "../../state";
+import { AppState, setAppState } from "../../state/app";
+import * as styles from "./styles";
+import { Text, Spinner } from "@klimadao/lib/components";
+import CloseIcon from "@mui/icons-material/Close";
+import { useSelector } from "react-redux";
+import { selectAppState } from "../../state/selectors";
+import { useWeb3 } from "@klimadao/lib/utils";
+
+type Props = {
+  address?: string;
+};
+
+export const BuyModal: FC<Props> = (props) => {
+  const dispatch = useAppDispatch();
+  const { buyModalService }: AppState = useSelector(selectAppState);
+
+  const [isLoadingMobilumWidget, setIsLoadingMobilumWidget] = useState(false);
+  const [modalHeight, setModalHeight] = useState<number | null>(null);
+
+  const web3 = useWeb3();
+
+  useEffect(() => {
+    if (buyModalService && buyModalService.length > 0) {
+      if (buyModalService === "mobilum") {
+        const _getWidget = async (): Promise<void> => {
+          setIsLoadingMobilumWidget(true);
+          const res = await fetch("/api/buy", {
+            method: "POST",
+            body: JSON.stringify({
+              address: props.address,
+            }),
+          });
+
+          const json = await res.json();
+
+          const element = document.getElementById("mobilumWidgetContainer");
+          if (element) {
+            setIsLoadingMobilumWidget(false);
+            element.innerHTML = json.html;
+            const script = document.createElement("script");
+            script.async = true;
+            script.src = json.scriptUrl;
+            element.appendChild(script);
+          }
+        };
+
+        _getWidget();
+      }
+    }
+  }, [buyModalService]);
+
+  const resizeHandler = useCallback(() => {
+    if (window.innerHeight < 800) {
+      console.log(window.innerHeight);
+      setModalHeight(window.innerHeight - 200);
```

Comment:
this is just viewport height, we can do it with plain css!
`calc(100vh - 20rem)`

This resize listener is buggy on windows, it doesn't always fire so I was able to create bugs where the modal stays smooshed. 
![image](https://user-images.githubusercontent.com/88635679/191605390-6d46ac5e-07d1-4560-97bf-9edae5680281.png)

Take another stab, I'm confident we can scrap any javascript styling and accomplish this layout with pure css

Reply:
Great! My CSS knowledge is stuck in 2008 or earlier ...